features/fixtures/mazerunner/src/main/java/com/bugsnag/android/mazerunner/scenarios/ManualSessionSmokeScenario.kt

```diff
@@ -0,0 +1,65 @@
+package com.bugsnag.android.mazerunner.scenarios
+
+import android.content.Context
+import android.os.Handler
+import android.os.HandlerThread
+import android.util.Log
+import com.bugsnag.android.*
+import com.bugsnag.android.JavaHooks.generateDelivery
+import com.bugsnag.android.flushAllSessions
+import java.lang.Thread
+
+/**
+ * Sends an exception after pausing the session
+ */
+internal class ManualSessionSmokeScenario(config: Configuration,
+                                          context: Context) : Scenario(config, context) {
+
+    init {
+        config.autoTrackSessions = false
+        val baseDelivery = createDefaultDelivery()
+
+        class InterceptingDelivery(private val baseDelivery: Delivery,
```

Comment:
It'd be good if this could be a top-level class that just invoked a callback with zero params, and if the `state` variable was stored in the scenario. This would allow the class to be reused by other scenarios - which should help reduce flakes.

Reply:
Moved this out into a separate class, seems to work as expected.